rfc/2022/2022-06-15 - The permissions lifecycle for platform tools wiill be managed via version control.md

```diff
@@ -0,0 +1,105 @@
+2. The application has been integrated into Keycloak, but has not configured authorization
+
+### 1: Describe Permissions
+This step encompasses the process of detailing a platform tool's permissions in JSON format<sub>[[RFC6]](https://vfs.atlassian.net/wiki/spaces/AMT/pages/2142011842/RFC+6+AMT+integrators+can+describe+fine-grained+permissions+using+JSON)</sub>, including any support from AMT.
+
+### 2: Apply Permissions
+The permissions application step includes the source control integration process. It also includes the automation required to apply the permissions configuration to the platform tool previously integrated into Keycloak. This is described in more detail below
+
+### 3: Use Permissions
+This step represents some amount of time where the integrated and configured platform tool's users interact with the tool under the scope of permissions they have been provided
+
+### 4: Revise Permissions
+Revising permissions could occur when:
+1. An authorization bug has been found and must be corrected
+2. A role must permanently or temporarily promote or demote its access to resources
+3. An endpoint/resource of a platform tool has been deprecated
+4. A platform tool has been deprecated or phased out
+   
+### Source code process
+```mermaid
+%%{init: { 'theme':'neutral' } }%%
+flowchart
+    subgraph permissions_lifecycle[ ]
+        direction LR
+        subgraph ptt[Platform Tool Team]
+            direction LR
+            auth_needed(Authorization change needed)
+            desc_perms(Describe permissions)
+            pr(Pull Request)
+
+            auth_needed --> desc_perms
+            desc_perms --> pr
+        end
+
+        pr --> review
+        
+        subgraph amt[Access Management Team]
+            direction LR
+            review(Review PR)
+            suggests{Suggestions?}
+            approve(Approve PR)
+            review --> suggests
+            suggests --> |Yes| pr
+            suggests --> |No| approve
+            approve --> gen_tf
+
+            subgraph auto[Automation]
+                direction TB
+                subgraph gha[Github Actions Workflow]
+                    gen_tf[Generate TF config from perms.json] --> gha_pr(Create PR for TF config)
+                end
+                
+
+                subgraph tf[Terraform/Keycloak]
+                    tf_apply[Terraform apply] --> kc_client_updated[Keycloak client permissions updated]
+                end
+            end
+           
+            gha_pr --> amt_tf_review[AMT review]
+            amt_tf_review --> suggests2{Suggestions?}
+            suggests2 --> |No| merge[AMT merge]
+            suggests2 --> |Yes| pr
+            merge --> tf_apply
+      end
+    end
+```
+
+## Risks
+Adding permissions in plain text (JSON) format to a repo has some security implications. A malicious actor could see and potentially do harm with role, resource, and scope information if they were given access to the Github organization.
```

Comment:
Maybe not needed, can we at least obfuscate the json data here ? E.g. maybe use something like base64 encoding.

Reply:
Good ideas @calquisola & @jchapin. One more idea is that we could potentially store the permissions files in the `va.gov-team-sensitive` repo. Though I've never done that before, so I'm not sure if that repo is intended for that kind of info. @td-usds might know.

Reply:
the `devops` repo is also private and contains IAM config